kayenta-web/src/main/java/com/netflix/kayenta/controllers/CanaryConfigController.java

```diff
@@ -111,14 +111,24 @@ public CanaryConfigUpdateResponse storeCanaryConfig(
     canaryConfig.setUpdatedTimestampIso(
         Instant.ofEpochMilli(canaryConfig.getUpdatedTimestamp()).toString());
 
-    String canaryConfigId = UUID.randomUUID() + "";
+    String canaryConfigId;
+
+    if (StringUtils.isEmpty(canaryConfig.getId())) {
+      canaryConfigId = UUID.randomUUID() + "";
+
+      // Ensure that the canary config id is stored within the canary config itself.
+      canaryConfig = canaryConfig.toBuilder().id(canaryConfigId).build();
+    } else {
```

Comment:
Probably just personal preference, but I think this would read easier if you removed the else block, so ti's more like

```
if (isEmpty(canaryConfig.getId())) {
  canaryConfig = canaryConfig.toBuilder().id(UUID.randomUUID().toString()).build();
}

String canaryConfigId = canaryConfig.getId();
```

Feel free to ignore if you disagree.

:+1: to not modifying the object more than we already do... it's certainly surprising to pass an object to a `store()` method and find out the method secretly also modified it.

Reply:
That does read easier. Changed it.